adev/src/app/core/services/version-manager.service.ts

```diff
@@ -17,39 +18,81 @@ export interface Version {
 
 export type VersionMode = 'stable' | 'deprecated' | 'rc' | 'next' | number;
 
-export const INITIAL_DOCS_VERSION = 17;
-export const VERSION_PATTERN_PLACEHOLDER = '{{version}}';
+export const INITIAL_ADEV_DOCS_VERSION = 17;
+export const VERSION_PLACEHOLDER = '{{version}}';
+export const MODE_PLACEHOLDER = '{{prefix}}';
 
 @Injectable({
   providedIn: 'root',
 })
 export class VersionManager {
+  private readonly window = inject(WINDOW);
+
+  // Note: We can assume that if the URL starts with v{{version}}, it is documentation for previous versions of Angular.
+  // Based on URL we can indicate as well if it's rc or next Docs version.
+  private readonly isDeprecated = this.window.location.hostname.startsWith(`v`);
+  private readonly isReleaseCandidate = this.window.location.hostname.startsWith(`rc`);
+  private readonly isNext = this.window.location.hostname.startsWith(`next`);
+
+  private get currentVersionMode(): VersionMode {
+    if (this.isDeprecated) return 'deprecated';
+    if (this.isReleaseCandidate) return 'rc';
+    if (this.isNext) return 'next';
+
+    return 'stable';
+  }
```

Comment:
I think we can just move the code that checks which version we use here (no need to pre-compute it):
```suggestion
    const hostname = this.window.location.hostname;
    if (hostname.startsWith('v')) return 'deprecated';
    if (hostname.startsWith('rc')) return 'rc';
    if (hostname.startsWith('next')) return 'next';

    return 'stable';
  }
```

Reply:
(this is **not a blocking comment**, just an optimization idea)